packages/next/src/cli/next-info.ts

```diff
@@ -124,4 +169,460 @@ const nextInfo: CliCommand = async (argv) => {
+            result: 'pass',
+          }
+        },
+      },
+    },
+    {
+      // For the simplicity, we only check the correctly installed optional dependencies -
+      // as this is mainly for checking DLOPEN failure. If user hit MODULE_NOT_FOUND,
+      // expect above next-swc installation would give some hint instead.
+      title: 'next-swc shared object dependencies',
+      scripts: {
+        linux: async () => {
+          const skipMessage =
+            'This diagnostics uses system-installed tools (ldd) to check next-swc dependencies, but it is not found. Skipping dependencies check.'
+
+          return await runSharedDependencyCheck(
+            [
+              {
+                bin: 'ldd',
+                checkArgs: ['--help'],
+                args: ['--verbose'],
+              },
+            ],
+            skipMessage
+          )
+        },
+        win32: async () => {
+          const skipMessage = `This diagnostics uses system-installed tools (dumpbin.exe) to check next-swc dependencies, but it was not found in the path. Skipping dependencies check.
+          dumpbin (https://learn.microsoft.com/en-us/cpp/build/reference/dumpbin-reference) is a part of Microsoft VC toolset,
+          can be installed with Windows SDK, Windows Build tools or Visual Studio.
+
+          Please make sure you have one of them installed and dumpbin.exe is in the path.
+          `
+
+          return await runSharedDependencyCheck(
+            [
+              {
+                bin: 'dumpbin.exe',
+                checkArgs: ['/summary'],
+                args: ['/imports'],
+              },
+            ],
+            skipMessage
+          )
+        },
+        darwin: async () => {
+          const skipMessage =
+            'This diagnostics uses system-installed tools (otools, dyld_info) to check next-swc dependencies, but none of them are found. Skipping dependencies check.'
+
+          return await runSharedDependencyCheck(
+            [
+              {
+                bin: 'otool',
+                checkArgs: ['--version'],
+                args: ['-L'],
+              },
+              {
+                bin: 'dyld_info',
+                checkArgs: [],
+                args: [],
+              },
+            ],
+            skipMessage
+          )
+        },
+      },
+    },
+  ]
+
+  // Collected output after running all tasks.
+  const report: Array<{
+    title: string
+    result: TaskResult
+  }> = []
+
+  console.log('\n')
+  for (const task of tasks) {
+    if (task.targetPlatform && task.targetPlatform !== currentPlatform) {
+      report.push({
+        title: task.title,
+        result: {
+          messages: undefined,
+          output: `[SKIPPED (${os.platform()} / ${task.targetPlatform})] ${
+            task.title
+          }`,
+          result: 'skipped',
+        },
+      })
+      continue
+    }
+
+    const taskScript = task.scripts[currentPlatform] ?? task.scripts.default!
```

Comment:
The `default!` here looks like it could be a bug.

In what scenario is there no default and could it happen here?

Reply:
There shouldn't be. There should be either platform-specific or default, it's just the interface of taskscript marks these as optional.